### PR TITLE
APPS-891 build without temp project

### PR DESCRIPTION
**Pull request by mhrvol**

### CODE REVIEW

File: src/python/dxpy/bindings/dxfile.py [hunk 1]
```diff
@@ -709,9 +709,6 @@ def get_upload_url_and_headers():
         # The file upload API requires us to get a pre-authenticated upload URL (and headers for it) every time we
         # attempt an upload. Because DXHTTPRequest will retry requests under retryable conditions, we give it a callback
         # to ask us for a new upload URL every time it attempts a request (instead of giving them directly).
-        # APPS-650 - retries are given because part would sometimes stay in non-complete state. We retry to reupload the part in case this happens.
```
Comment (mhrvol) [inline src/python/dxpy/bindings/dxfile.py:712]: previous PR removed code related to this. Removing comment and redundant code.

File: Readme.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ toolkit that doesn't require them.
 
 ### Ubuntu 20.04
 
-    sudo apt install python-is-python3 python3-venv libssl-dev flex bison
+    sudo apt install python-is-python3 python3-venv libssl-dev flex bison libffi-dev
```
Comment (commandlinegirl) [inline Readme.md:58]: Why do we need this?

**Review by commandlinegirl — CHANGES_REQUESTED** [review]
Please add a validation that checks if the app is a single-region app. If it is not, error out (see Jira ticket description point 1 for info).

File: Readme.md (hunk 2 above)
Comment (mhrvol) [reply Readme.md:58]: This library is missing when we try to install dx-toolkit on our latest cloud_workstation with ubuntu 20.04.

When I previously created this readme,  I used slightly different environment that already had this library.

Comment (commandlinegirl) [reply Readme.md:58]: Got it, thanks

**Comment by mhrvol** [comment]
> Please add a validation that checks if the app is a single-region app. If it is not, error out (see Jira ticket description point 1 for info).

This is already in place:

```
if enabled_regions is not None and len(enabled_regions) > 1 and not use_temp_build_project:
```

File: src/python/dxpy/bindings/dxfile.py (hunk 1 above)
Comment (commandlinegirl) [reply src/python/dxpy/bindings/dxfile.py:712]: OK

**Comment by commandlinegirl** [comment]
> > Please add a validation that checks if the app is a single-region app. If it is not, error out (see Jira ticket description point 1 for info).
> 
> This is already in place:
> 
> ```
> if enabled_regions is not None and len(enabled_regions) > 1 and not use_temp_build_project:
> ```

Oh, true, thanks for the pointer!

**Review by commandlinegirl — APPROVED** [review]
LGTM!

**Pull request merged**